### ISSUE TITLE
Bot runs through A5 after doing Diablo Walk

### ISSUE THREAD

**Issue by Buttpickle** [kolton/d2bot-with-kolbot#3145]
Crazy bot runs out of town through the Act, destination unknown after completing the Diablo walk just fine. "Trying to get the waypoint" Trying to go back to autobaal now and will give the other character all of the WPs out of curiosity. 

Been changing stuff here and there but nothing seems to work. I assume it's a simple solution and I'm overlooking it.

I'll paste my configs;

Follower Character Config
https://pastebin.com/w9fi7dWu

Leader Character Config
https://pastebin.com/6qk2pg27

D2BotFollow
https://pastebin.com/LKa4jbki

D2BotLead
https://pastebin.com/4LymZiUx

**Comment by DarkHorseDre** [comment]
when this happens, press the home key to see the console then print screen when it goes walkabout so we see what the bot is trying to do
(I'm guessing the next script is trying to get so a place where you don't have the wp)

**Comment by Buttpickle** [comment]
Thanks for the response, it was, "Trying to get the waypoint".

I gave the bot all the WPs in A5 and it behaves as it should.. It doesn't take any WPs, it just stands in town and waits for the portal. Very strange..

**Comment by DarkHorseDre** [comment]
yeah if your boss script tells the bot to go somewhere by wp and it doesnt have it, it will try and travel (walk or tp) there.
e.g. Baal script starts with moving to WSK2 `Pather.useWaypoint(129);` 

> It doesn't take any WPs

when??

> it just stands in town and waits for the portal.

which one does? the leader? which script is it configured to be using at that time? (you will see it in the console)

**Comment by Buttpickle** [comment]
> yeah if your boss script tells the bot to go somewhere by wp and it doesnt have it, it will try and travel (walk or tp) there.
> e.g. Baal script starts with moving to WSK2 `Pather.useWaypoint(129);`
> 
> > It doesn't take any WPs
> 
> when??
> 
> > it just stands in town and waits for the portal.
> 
> which one does? the leader? which script is it configured to be using at that time? (you will see it in the console)

The Leecher doesn't take any WPs, ever--or at least I don't think it's supposed to. Unless of course, traveling from town to town to wait for a portal. Which is why I thought it was odd that the Leecher needed a WP to work when it's just supposed to take the Leader's portal.

The Leader behaves normally and I got the Leacher to work fine by giving him all the WPs in A5. Oddly, it doesn't have any A4 WPs and does a Diablo walk just fine. I guess it's the Baal Leech script.

**Comment by DarkHorseDre** [comment]
Well I still don't have enough info from you, but from what you've said the leecher is running the baal leech script or possibly another?

Find out the script the follower is running and search it for that pather.usewaypoint line - you may see it is trying to use a wp.

if you had followed my advice to use the console it would have told you the area id the bot was trying to get to.
the script it is running and the area it is trying to get to would have answered your questions.

spend time to understand what the boss does and how - read the config scripts and the bot scripts for keywords and it will make sense.
